ffi/src/sspi/sec_pkg_info.rs

```diff
@@ -22,8 +22,10 @@ pub struct SecPkgInfoW {
 
 pub type PSecPkgInfoW = *mut SecPkgInfoW;
 
+pub struct RawSecPkgInfoW(pub *mut SecPkgInfoW);
```

Comment:
It was fun to investigate and fix. In short, conversions chain like this one `*mut SecPkgInfoW -> &mut SecPkgInfoW -> *mut SecPkgInfoW` can create UB 😮 
![image_2024-07-25_22-31-46](https://github.com/user-attachments/assets/eb38d928-8f81-4e35-946a-64a9b9123440)
When we create `&mut SecPkgInfoW`, Rust creates a SharedReadWrite tag (it relates to stacked borrows and so on). And when we convert it to the raw pointer and try to deallocate, we have UB

Reply:
Good job!